### ISSUE TITLE
Error after terraform apply due to variable name change?

### ISSUE THREAD

**Issue by koenijn** [hashicorp/terraform#7795]
Hello all,

After renaming a variable name in a `template_file`-resource's template, terraform apply finishes with the error mentioned below. However the execution of the plan seems to be ok according to the logging output and a manual inspection.

Error applying plan:

1 error(s) occurred:
- aws_s3_bucket_object.install_ede_ps1: diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.

Please include the following information in your report:

```
Terraform Version: 0.6.16
Resource ID: aws_s3_bucket_object.install_ede_ps1
Mismatch reason: attribute mismatch: content
Diff One (usually from plan): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"content":*terraform.ResourceAttrDiff{Old:"# Create directories\r\nNew-Item -ItemType Directory -Force -Path c:\\philips\\cheetah\\ede\\node\\\r\nNew-Item -ItemType Directory -Force -Path c:\\philips\\cheetah\\prereqs\\\r\nNew-Item -ItemType Directory -Force -Path c:\\philips\\cheetah\\ede\\db\\\r\n\r\n# Version of current EDE file\r\n$EDE = \"22.0.0.2995\"\r\n$LOCAL_EDE_FILE = \"c:\\philips\\cheetah\\prereqs\\EDE-22.0.0.2995.zip\"\r\n\r\n# Copy file from s3 to the location machine\r\nRead-S3Object -BucketName \"cheetah-060712730170-eu-west-1-ede-binaries\" -Key \"archive/EDE-22.0.0.2995.zip\" -File $LOCAL_EDE_FILE\r\nRead-S3Object -BucketName \"cheetah-060712730170-eu-west-1-ede-binaries\" -Key \"tf-aachen/prereqs/EDE_config.zip\" -File \"c:\\philips\\cheetah\\prereqs\\EDE_config.zip\"\r\nRead-S3Object -BucketName \"cheetah-060712730170-eu-west-1-ede-binaries\" -Key \"tf-aachen/prereqs/fnr.exe\" -File \"c:\\philips\\cheetah\\prereqs\\fnr.exe\"\r\nRead-S3Object -BucketName \"cheetah-060712730170-eu-west-1-ede-binaries\" -Key \"tf-aachen/prereqs/set_aws_private_ip.bat\" -File \"c:\\philips\\cheetah\\prereqs\\set_aws_private_ip.bat\"\r\nRead-S3Object -BucketName \"cheetah-060712730170-eu-west-1-ede-binaries\" -Key \"tf-aachen/prereqs/start_ede.bat\" -File \"C:\\Users\\Administrator\\Desktop\\start_ede-22.0.0.2995.bat\"\r\n\r\n# Unzip EDE\r\nset-alias 7z \"$env:ProgramFiles\\7-Zip\\7z.exe\"\r\n7z x $LOCAL_EDE_FILE -oc:\\philips\\cheetah\\ede -y\r\n\r\n# Unzip EDE configuration\r\nset-alias 7z \"$env:ProgramFiles\\7-Zip\\7z.exe\"\r\n7z x \"c:\\philips\\cheetah\\prereqs\\EDE_config.zip\" -oc:\\philips\\cheetah\\ede -y\r\n\r\n# Update EDE configuration with EDE specific IPs\r\nSet-Location -Path c:\\philips\\cheetah\\prereqs\\\r\nStart-Process -Wait -NoNewWindow -FilePath c:\\philips\\cheetah\\prereqs\\set_aws_private_ip.bat\r\n\r\n# Update the firewall\r\nnetsh advfirewall firewall set rule group=\"remote desktop\" new enable=Yes\r\nnetsh advfirewall firewall add rule name=\"Allow icebox.exe\" dir=in action=allow program='C:\\Program Files (x86)\\ZeroC\\Ice-3.6.2\\bin\\vc110\\x64\\icebox.exe'\r\nnetsh advfirewall firewall add rule name=\"EDE1\" protocol=TCP dir=in profile=any localport=10020 remoteip=any localip=any action=allow\r\nnetsh advfirewall firewall add rule name=\"EDE2\" protocol=TCP dir=in profile=any localport=11000 remoteip=any localip=any action=allow\r\n", New:"${template_file.install_ede_ps1.rendered}", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}}, Destroy:false, DestroyTainted:false}
Diff Two (usually from apply): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff(nil), Destroy:false, DestroyTainted:false}
```

Also include as much context as you can about your config, state, and the steps you performed to trigger this error.

**Comment by stack72** [comment]
Hi @koenijn 

Apologies this is causing errors for you. Please can you provide a basic config that will allow us to reproduce this error? This will help us to debug

Thanks

Paul

**Comment by mitchellh** [comment]
Hello! This should be fixed in later versions of Terraform (0.7.11 at time of writing). 

Its hard to know 100% since there isn't a clear reproduction case here, but we've fixed issues with this exact error message (and type) in recent versions. Please try again. Thanks!
